invenio_files_rest/models.py

```diff
@@ -298,12 +299,43 @@ def get(cls, bucket_id):
         ).one_or_none()
 
     @classmethod
+    def exists(cls, bucket_id):
+        """Check if the given bucket exists.
+
+        :param bucket_id: Bucket identifier.
+        :returns: Boolean.
+        """
+        return cls.get(bucket_id) is not None
+
+        # db.session.query(
+        #    exists().where(
+        #        cls.id == bucket_id and cls.deleted is False
+        #    )).scalar()
+
+    @classmethod
     def all(cls):
         """Return query of all buckets (excluding deleted)."""
         return cls.query.filter_by(
             deleted=False
         )
 
+    @classmethod
+    def delete(cls, bucket_id):
+        """Delete a bucket.
+
+        Does not actually delete the Bucket, just marks it as deleted.
+        Deleting a bucket also marks the contained objects as deleted.
+        """
+        bucket = cls.get(bucket_id)
+        if not bucket or bucket.deleted:
+            return False
+
+        for obj in ObjectVersion.query.filter_by(bucket_id=bucket_id):
```

Comment:
I'm not sure if we should mark the objects as deleted. Say you want to undelete an entire bucket.